packages/happy-dom/src/nodes/html-option-element/HTMLOptionElement.ts

```diff
@@ -67,10 +81,26 @@ export default class HTMLOptionElement extends HTMLElement implements IHTMLOptio
 	 * @param selected Selected.
 	 */
 	public set selected(selected: boolean) {
-		if (!selected) {
-			this.removeAttributeNS(null, 'selected');
-		} else {
-			this.setAttributeNS(null, 'selected', '');
```

Comment:
Is this intentional that you're not adding/removing the `selected` DOM attribute anymore? Isn't that necessary? This'll make it harder to implement multi-selects in the future too I'm guessing.

Reply:
Changing the `selectedIndex` property should not modify the attribute value. It should set the `HTMLOptionElement` to be dirty and then it overrides the attribute.

It is not entirely correct now as I missed handling the attribute, sorry about that.

https://html.spec.whatwg.org/multipage/form-elements.html#dom-select-selectedindex
https://html.spec.whatwg.org/multipage/form-elements.html#concept-option-dirtiness

Reply:
Thanks! I assumed setting "selectedness" means setting the selected attribute, which does seem wrong. I can't find anything on those links that specifically details what happens when an option's selectedness is set, which is strange, but what you're saying definitely makes sense